llvm/lib/Transforms/Utils/LoopUnroll.cpp

```diff
@@ -752,6 +752,14 @@ llvm::UnrollLoop(Loop *L, UnrollLoopOptions ULO, LoopInfo *LI,
         }
       }
 
+      // Remap source location atom instance. Do this now, rather than
+      // when we remap instructions, because remap is called once we've
+      // cloned all blocks (all the clones would get the same atom
+      // number).
```

Comment:
I get why all the clones need to get distinct atom numbers; what is it here that ensures that's the case, don't we need to reset something in VMap to ensure that's the case? i.e., if we called RemapSourceAtom later what would be different?

(I get the overall picture, but not how it's achieved here.)

Reply:
Ah right -- it's the fact that Something Happens (TM) during CloneBasicBlock that I think I was missing.